src/benchmark.rs

```diff
@@ -0,0 +1,64 @@
+use crate::lang_impl::LangImpl;
+
+use std::collections::HashMap;
+
+/// The key of the path tag.
+pub const TAG_PATH: &str = "path";
+
+/// A collection of tags associated with a benchmark.
+///
+/// A tag is a key-value pair. It records both arbitrary values set by the user,
+/// and the results of a benchmark.
+pub type TagStore = HashMap<String, String>;
+
+/// A benchmark, which consists of a set of tags, and a list of language
+/// implementations the benchmark will be run on.
+pub struct Benchmark<'a> {
+    tags: TagStore,
+    lang_impls: Vec<&'a dyn LangImpl>,
```

Comment:
Is this right? I though (perhaps erroneously) that you'd generate one benchmark per different language implementation?

Reply:
I thought we wanted to have multiple language implementations for each benchmark. Do you think it would be better to have a one-to-one mapping between `Benchmark`s and `LangImpl`s instead?

Reply:
Having thought about this a bit, I think a 1:1 mapping does have the advantage of simplicity.